front/src/App.js

```diff
@@ -156,13 +146,24 @@ class App extends Component {
    * @param onIncorrectToken Callback from AuthenticationScreen to update the
    * AuthenticationScreen in case the token was incorrect.
    */
-  onSubmit(token, onIncorrectToken) {
-    //TODO: Implement a correct version
-    if (token === "techunited") {
-      this.setState({ loggedIn: true, token: token });
-    } else {
-      onIncorrectToken();
-    }
+  authSubmit(token, onIncorrectToken) {
+    const l = window.location;
+    console.log(`send authorization to ${l.protocol}//${l.host}/api/v1/auth`);
+    fetch(`${l.protocol}//${l.host}/api/v1/auth`, {
+      method: "GET",
+      headers: new Headers({
+        Authorization: "Basic " + btoa(`user:${token}`)
+      })
+    }).then(response => {
+      if (!response.ok) {
+        onIncorrectToken(response.status);
```

Comment:
Why would `onIncorrectToken` be a separate function if it's only ever called once in the code? Just put the logic from that function here and remove the callback altogether, you don't need it.

style nit: it would be clearer if you just put a return after the `onIncorrectToken` call and skip the `else`. That way the error handling would be indented and the "normal" execution would not.

Reply:
second thing: done. First thing, I'm not sure how to do that without a callback function, as I can't set the state of AuthenticationScreen from App.js.

Reply:
and the function is asynchronous, so just returning is not enough.